File: backEnd/src/main/java/com/smart/safais/model/Booking.java
package com.smart.safais.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private User customer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "driver_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private User driver;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "route_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Route route;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private Status status = Status.PENDING;

    private LocalDateTime scheduledTime;

    @Column(name = "created_at")
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    public enum Status {
        PENDING, ACCEPTED, ON_WAY, IN_PROGRESS, DECLINED, COMPLETED, PAID
    }
} 

File: frontEnd/src/app/model/Booking.ts
import { User } from './User';
import { Route } from './Route';

export interface Booking {
  id?: number;
  customer: User;
  driver?: User;
  route: Route;
  status: 'PENDING' | 'ACCEPTED' | 'ON_WAY' | 'IN_PROGRESS' | 'DECLINED' | 'COMPLETED' | 'PAID';
  scheduledTime: string;
  createdAt?: string;
}

export interface BookingResponse {
  success: boolean;
  message: string;
  data: Booking[];
} 